packages/flutter_tools/lib/src/asset.dart

```diff
@@ -1032,16 +1111,22 @@ class ManifestAssetBundle implements AssetBundle {
 class _Asset {
   const _Asset({
     required this.baseDir,
+    Uri? originUri,
     required this.relativeUri,
     required this.entryUri,
     required this.package,
     this.assetKind = AssetKind.regular,
-  });
+    this.flavor,
+  }): originUri = originUri ?? entryUri;
 
   final String baseDir;
 
   final Package? package;
 
+  /// The platform-independent URL provided by the user in the pubspec that this
+  /// asset was found from.
+  final Uri originUri;
```

Comment:
What is this being used for?

Reply:
See https://github.com/flutter/flutter/pull/132985#discussion_r1383401264. In short, it's to help disambiguate in scenarios where multiple declarations affect the same asset:

```yaml
assets:
  - path: assets/paid/
    flavor: paid
  - path: assets/paid/pip.mp3 # For pip.mp3, this declaration overrides the previous one.
    flavor: free
```

This could probably just be a `ToolExit` instead though.